tasks/appraisal.rake

```diff
@@ -0,0 +1,187 @@
+      cmd = []
+      cmd << ['rm', '-vf', 'Gemfile.lock']
+      cmd << ['gem', 'install', 'bundler', '-v', bundler_version(ruby_version)] if bundler_version(ruby_version)
+      cmd << [*bundle(ruby_version), 'config', 'without', 'check']
+      cmd << [*bundle(ruby_version), 'install']
+      cmd << [*bundle(ruby_version), 'exec', 'appraisal', 'generate']
+      cmd << [*bundle(ruby_version), 'exec', 'appraisal', 'install']
+      cmd << ['rm', '-vf', 'Gemfile.lock']
+
+      cmd = cmd.map { |c| c << '&&' }.flatten.tap(&:pop)
+
+      p cmd
+      p docker(ruby_version, cmd)
+      sh docker(ruby_version, cmd).join(' ')
+    end
+
+    ENV.delete('APPRAISAL_SKIP_BUNDLE_CHECK')
+  end
+
+  desc 'Update Appraisal gemfiles. Takes a version list as argument, defaults to all'
+  task :update do |_task, args|
+    ruby_versions(args.to_a).each do |ruby_version|
+      cmd = []
+      cmd << ['rm', '-vf', 'Gemfile.lock']
+      cmd << ['gem', 'install', 'bundler', '-v', bundler_version(ruby_version)] if bundler_version(ruby_version)
+      cmd << [*bundle(ruby_version), 'config', 'without', 'check']
+      cmd << [*bundle(ruby_version), 'install']
+      cmd << [*bundle(ruby_version), 'exec', 'appraisal', 'generate']
+      cmd << [*bundle(ruby_version), 'exec', 'appraisal', 'update']
+      cmd << ['rm', '-vf', 'Gemfile.lock']
+
+      cmd = cmd.map { |c| c << '&&' }.flatten.tap(&:pop)
+
+      p cmd
+      p docker(ruby_version, cmd)
+      sh docker(ruby_version, cmd).join(' ')
+    end
+  end
+
+  desc 'Add all platforms to Appraisal gemfiles. Takes a version list as argument, defaults to all'
+  task :platform do |_task, args|
+    ruby_versions(args.to_a).each do |ruby_version|
+      next if ruby_version.start_with?('jruby-')
+
+      cmd = []
+      cmd << ['rm', '-vf', 'Gemfile.lock']
+      cmd << ['gem', 'install', 'bundler', '-v', bundler_version(ruby_version)] if bundler_version(ruby_version)
+      cmd << [*bundle(ruby_version), 'config', 'without', 'check']
+      p lockfile_prefix(ruby_version)
+      Dir["gemfiles/#{lockfile_prefix(ruby_version)}_*.gemfile.lock"].each do |lockfile|
+        gemfile = lockfile.gsub(/\.lock$/, '')
+        cmd << ['env', "BUNDLE_GEMFILE=#{gemfile}",
+                *bundle(ruby_version), 'lock',
+                '--lockfile', lockfile,
+                '--add-platform', 'x86_64-linux']
+        cmd << ['env', "BUNDLE_GEMFILE=#{gemfile}",
+                *bundle(ruby_version), 'lock',
+                '--lockfile', lockfile,
+                '--add-platform', 'aarch64-linux']
+      end
+      cmd << ['rm', '-vf', 'Gemfile.lock']
+
+      cmd = cmd.map { |c| c << '&&' }.flatten.tap(&:pop)
+
+      p cmd
+      p docker(ruby_version, cmd)
+      sh docker(ruby_version, cmd).join(' ')
+    end
+  end
+end
+
+TRACER_VERSIONS = %w[
+  2.1
+  2.2
+  2.3
+  2.4
+  2.5
+  2.6
+  2.7
+  3.0
+  3.1
+  3.2
+  jruby-9.2.8.0
+  jruby-9.2
+  jruby-9.3
+  jruby-9.4
+  #truffleruby-22.3.0
```

Comment:
This breaks on my machine -- did you intend to add code to skip truffleruby? Or perhaps it's easier to just remove it for now -- it's not like it's enabled in CI anyway, so perhaps rather than being a zombie we should just clean this up and re-add it later when we're ready.

Reply:
Yeah we should remove it from the list. I don't think I was ever able to generate lockfiles for truffleruby, even before this PR.

Reply:
Actually, I've been thinking I'll open a PR to remove TruffleRuby completely for now. At some point it looked like we were really close to adding it, but nobody had time since to pick this up, so I'll just rip it out rather than leaving it in this half-zombie state, since we're doing ongoing work but not getting anything in return.